### ISSUE TITLE
a comma in an address will cause troubles

### ISSUE THREAD

**Issue by tgirod** [pazz/alot#773]
A friend have an address `SURNAME, Firstname <address@domain.com>`

When replying to him, this address will go into the `To:` header field, unescaped. With the comma, msmtp will understand this as two emails : `SURNAME` and `Firstname <address@domain.com>`). Because the first address is invalid, it won't work.

I didn't take the time to check the RFC, but when replying, I think the field should be `"SURNAME, Firstname" <address@domain.com>`.

**Comment by pazz** [comment]
I agree, this sounds like a bug and should be fixed by searching for a comma in the header values
somewhere around here: https://github.com/pazz/alot/blob/master/alot/commands/thread.py#L184

**Comment by proycon** [comment]
I've been running into this same issue on several occasions.

**Comment by lfos** [comment]
I cannot reproduce this. If names appearing in the From header contain commas, they need to be quoted as well, so I don't see why we would need to escape them once more.

**Comment by proycon** [comment]
The problem arises precisely when the names are unquoted, for whatever reason that may be. I suppose an extra validation step that ensures everything in the To/Cc/Bcc fields at least looks like it contains an e-mail address will solve it already.

**Comment by lfos** [comment]
Well, the thing is that RFC 5322 explicitly allows multiple From addresses. While that feature probably isn't used a lot, working around the issue you described would mean that we would handle such well-formed emails incorrectly (according to the standards). Instead, you (or your friend) should try to figure out why those emails are malformatted and fix that.

**Comment by tgirod** [comment]
@lfos I think we were talking about addresses with commas in destination fields, not origin field.

**Comment by lfos** [comment]
Maybe I misunderstood you but I thought you were talking about an email that has a From header of the form `SURNAME, Firstname <address@domain.com>`. If you reply to that email, the To header has the same form (i.e. two separate recipients).

The problem isn't within alot but rather that the From header of the email you are replying to is malformed already. It should look like `"SURNAME, Firstname" <address@domain.com>` (note the double quotes). If the double quotes are missing, replying to two authors `SURNAME` and `Firstname <address@domain.com>` is the correct thing to do according to the standards.

**Comment by lucc** [comment]
There has been some change to the code that handles address lists in headers I think. Is this still happening and if so are there any valid arguments against @lfos (I think he is right). If not we can close this.

**Comment by tgirod** [comment]
Don't know, haven't been using alot for a long time
